optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/move/Move.java

```diff
@@ -87,6 +87,18 @@
      */
     Move<Solution_> doMove(ScoreDirector<Solution_> scoreDirector);
 
+    /**
+     * As defined by {@link #doMove(ScoreDirector)}, but does not return an undo move.
+     * It is recommended that {@link #doMove(ScoreDirector)} is implemented by wrapping {@link #doMoveOnly(ScoreDirector)}.
```

Comment:
This sentence confuses me, especially with the implementation.

In general, we recommend users to implement AbstractMove and only implement doGenuineMove there. Let's keep it simple and recommend that and only that.

CompositiveMove cannot extend AbstractMove because it needs to sequence undo move creation and do move execution correctly. But for that one exception I don't think we need to confuse the javadocs, Wdyt?

Reply:
```suggestion
```